### PR TITLE
[FIX] Employee can select future payment date in edit popup 

### PR DESCRIPTION
**Pull request by Andrii-Danylenko**
## GIT

* [Main GIT ticket](https://github.com/ita-social-projects/GreenCity/issues/8200)

I have introduced new SettlementDate validator to ensure that passed date is not in the future. 
Also fixed old tests which required validator

## Summary by CodeRabbit

- **New Features**
  - Enhanced input validation for order management and manual payment processing.
  - Enforced stricter rules to ensure settlement dates are not set in the future.
  - Introduced a new method for constructing `CertificateDtoForAdding`.

- **Bug Fixes**
  - Updated date handling in `ManualPaymentRequestDto` to ensure correct formatting.
  - Corrected method references to align with updated naming conventions.

- **Tests**
  - Added tests to verify that attempts to use future settlement dates are correctly rejected by the system.
  - Introduced a new test class for validating settlement date functionality.
  - Updated test setup for improved efficiency and organization.

### CODE REVIEW

**Review by maks741 — APPROVED** [review]

File: core/src/test/java/greencity/ModelUtils.java [hunk 1]
```diff
@@ -283,7 +284,7 @@ public static ViolationDetailInfoDto getViolationDetailInfoDto() {
     public static ManualPaymentRequestDto getRequestDto() {
         return ManualPaymentRequestDto.builder()
             .amount(500L)
-            .settlementdate("09-02-2021")
+            .settlementdate("2021-03-07")
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/ModelUtils.java]: please follow the Java naming convention

File: service-api/src/main/java/greencity/validator/SettlementDateValidator.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import greencity.annotations.ValidSettlementDate;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class SettlementDateValidator implements ConstraintValidator<ValidSettlementDate, String> {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/validator/SettlementDateValidator.java:11]: We have different date formats for `ua` and `en` locales
Shouldn't we support both of them?

File: service-api/src/test/java/greencity/validator/SettlementDateValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        LocalDate localDate = LocalDate.now();
+        assertTrue(validator.isValid(formatter.format(localDate), context));
+        localDate = LocalDate.of(2020, 1, 1);
+        assertTrue(validator.isValid(formatter.format(localDate), context));
+        localDate = LocalDate.now().minusDays(1);
+        assertTrue(validator.isValid(formatter.format(localDate), context));
+    }
+
+    @Test
+    public void validateInvalidSettlementDate() {
```
Comment (RomaMocherniuk) [inline service-api/src/test/java/greencity/validator/SettlementDateValidatorTest.java:29]: please add test cases for `null` and the wrong format

File: service-api/src/main/java/greencity/validator/SettlementDateValidator.java (hunk 2 above)
Comment (Andrii-Danylenko) [reply service-api/src/main/java/greencity/validator/SettlementDateValidator.java:11]: I use data from the frontend, which is transmitted through the calendar, the date format is always the same, regardless of the locale.

File: service-api/src/main/java/greencity/dto/payment/ManualPaymentRequestDto.java [hunk 4]
```diff
@@ -16,6 +17,7 @@
 @Builder
 public class ManualPaymentRequestDto {
     @NotEmpty
+    @ValidSettlementDate
     private String settlementdate;
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/payment/ManualPaymentRequestDto.java]: please use camel case for the variable naming `private String settlementDate;`

**Review by RomaMocherniuk — APPROVED** [review]

File: service-api/src/main/java/greencity/validator/SettlementDateValidator.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public boolean isValid(String s, ConstraintValidatorContext constraintValidatorContext) {
```
Comment (VitaliiChernenko) [inline service-api/src/main/java/greencity/validator/SettlementDateValidator.java]: Is it possible to rename `String s` into something more understandable?

**Review by ChernenkoVitaliy — APPROVED** [review]

**Comment by ChernenkoVitaliy** [comment]
Cause you renamed `settlementdate` to `settlementDate` don't forget after merging to notice UI team cause in this case it also should be renamed on UI.

**Review by oleksandra-bulhakova — APPROVED** [review]
